File: official/docs/golang/current/smartrate/retrieve-standalone-smartrate-deliver-on.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v5"
)

func retrieveStandaloneSmartrateDeliverOn() {
	client := easypost.New("EASYPOST_API_KEY")

	params := &easypost.RecommendShipDateForZipPairParams{
		FromZip:             "10001",
		ToZip:               "10002",
		Carriers:            []string{"UPS", "FedEx"},
		DesiredDeliveryDate: "2024-07-18",
	}

	recommendations, _ := client.RecommendShipDateForZipPair(params)

	fmt.Println(recommendations)
}

File: official/docs/csharp/current/smartrate/retrieve-standalone-smartrate-deliver-on.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Newtonsoft.Json;
using EasyPost;
using EasyPost.Models.API;
using EasyPost.Parameters;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            Parameters.SmartRate.RecommendShipDateForZipPair recommendShipDateForZipPairParameters = new()
            {
                OriginPostalCode = address1Parameters.Zip,
                DestinationPostalCode = address2Parameters.Zip,
                DesiredDeliveryDate = Fixtures.DesiredDeliveryDate,
                Carriers = ["USPS"],
            };

            RecommendShipDateForZipPairResult results = await client.SmartRate.RecommendShipDate(recommendShipDateForZipPairParameters);

            Console.WriteLine(JsonConvert.SerializeObject(results, Formatting.Indented));
        }
    }
}
